futures-util/src/future/try_join_all.rs

```diff
@@ -83,54 +99,87 @@ where
 /// assert_eq!(try_join_all(futures).await, Err(2));
 /// # });
 /// ```
-pub fn try_join_all<I>(i: I) -> TryJoinAll<I::Item>
+pub fn try_join_all<I>(iter: I) -> TryJoinAll<I::Item>
 where
     I: IntoIterator,
-    I::Item: TryFuture,
+    I::Item: TryFuture
+        + Future<Output = Result<<I::Item as TryFuture>::Ok, <I::Item as TryFuture>::Error>>,
```

Comment:
This extra bound is needed to get `try_collect` to work.

Reply:
I think this can be worked around this by calling [into_future](https://docs.rs/futures/latest/futures/future/trait.TryFutureExt.html#method.into_future).